types/enzyme/index.d.ts

```diff
@@ -92,6 +92,14 @@ export interface CommonWrapper<P = {}, S = {}, C = Component<P, S>> {
      */
     hasClass(className: string | RegExp): boolean;
 
+    /**
+     * Invokes a function prop.
+     * @param invokePropName The function prop to call.
+     * @param ...args The argments to the invokePropName function
+     * @returns The value of the function.
+     */
+    invoke<K extends keyof P>(invokePropName: K): P[K];
```

Comment:
`invoke` returns a function, which when again invoked, calls the prop function.

Thus, the type here should perhaps be (assuming P is the props):
```suggestion
    invoke<K extends keyof P>(invokePropName: K): P[K] extends Function
```
but ideally it'd also give you a type error when you passed a `K` such that `P[K]` isn't a function.

Reply:
The return type isn't currently a valid position for a constraint in TS. This seems to do what we want

```ts
type FunctionPropertyNames<T> = { [K in keyof T]: T[K] extends Function ? K : never }[keyof T];

interface Wrapper<P> {
    invoke<K extends keyof Pick<P, FunctionPropertyNames<P>>>(invokePropName: K): P[K]
}
```

Reply:
Haha I was just working to update to use `FunctionPropertyNames`! Found that in the TS Handbook after a colleague at work mentioned it.  

This is what I had:
```typescript
/**
   * Invokes a function prop.
   * @param invokePropName The function prop to call.
   * @param ...args The argments to the invokePropName function
   * @returns The value of the function.
   */
  invoke<K extends NonNullable<{
      [K in keyof P]: P[K] extends () => void ? K : never
  }[keyof P]>>(invokePropName: K): P[K];
```

I inlined the `FunctionPropertyNames` because it seems that DT auto exports all declarations, so I wasn't sure if this was an export that should be owned by `@types/enzyme`. 

I was thinking of using `NonNullable` since the conditional allows for `undefined` and the inlined `FunctionPropertyNames` already "picks" those keys. I hadn't considered `Pick`. However, I do think `Pick` might make the intent more clear.  Thoughts on which to use?

Reply:
👍For inlining